src/Raven.Server/Rachis/Elector.cs

```diff
@@ -357,5 +373,21 @@ private HandleVoteResult ShouldGrantVote(TransactionOperationContext context, lo
 
             return result;
         }
+
+        public void Dispose()
+        {
+            if (_electionWon == false)
+                _connection.Dispose();
+
+            if (_engine.Log.IsInfoEnabled)
+            {
+                _engine.Log.Info($"{ToString()}: Disposing");
+            }
+
+            if (_electorLongRunningWork != null && _electorLongRunningWork.ManagedThreadId != Thread.CurrentThread.ManagedThreadId)
+                _electorLongRunningWork.Join(int.MaxValue);
+
+            _engine.InMemoryDebug.RemoveRecorderOlderThan(DateTime.UtcNow.AddMinutes(-5));
```

Comment:
why do we need this?

Reply:
Copied from Follower, not needed, but it is there as well.